src/cosmoscout/ObserverNavigationNode.cpp

```diff
@@ -134,33 +134,42 @@ bool ObserverNavigationNode::DoEvalNode() {
     vOffset  = glm::dvec3(tmp[0], tmp[1], tmp[2]);
   }
 
-  auto&      oObs              = mSolarSystem->getObserver();
-  glm::dvec3 vObserverPosition = oObs.getAnchorPosition();
-  glm::dquat qObserverRotation = oObs.getAnchorRotation();
-  double     dObserverScale    = oObs.getAnchorScale();
-
   auto vTranslation = mLinearDirection;
 
   vTranslation.x *= mMaxLinearSpeed[0];
   vTranslation.y *= mMaxLinearSpeed[1];
   vTranslation.z *= mMaxLinearSpeed[2];
-  vTranslation *= dObserverScale;
   vTranslation *= mLinearSpeed.get(dTtime);
 
   vTranslation *= dDeltaTime;
-  vTranslation += vOffset * dObserverScale;
+  vTranslation += vOffset;
+
+  auto&  oObs     = mSolarSystem->getObserver();
+  double stepSize = glm::length(vTranslation);
+
+  if (stepSize > 0.0) {
+    // Ensure that an SolarSystem::updateSceneScale() is called at least at 100 Hz. If it is called
+    // only once a frame, it can happen that the observer instantly travels to a planet's surface.
+    int32_t steps = std::ceil(dDeltaTime / 0.01);
```

Comment:
Why don't you simply multiply by 100? This would be a more in line with the 100 Hz in the comment above and it would also be more efficient (although that doesn't matter here).